client/script.js

```diff
@@ -0,0 +1,184 @@
+         method: "DELETE"
+         }).then(res => {
+          if (res.ok){
+		  deleteTodo(taskid);
+		  }
+          });
+         });
+        });
+  }
+	 
+	function toggleCompleteTodo(id) {
+  const taskElement = document.querySelector(`li[id='${id}']`);
+  if (taskElement.classList.contains("completed")) {
+    taskElement.classList.remove('completed');
+  } else {
+    taskElement.classList.add('completed');
+  }
+  
+}
+	
+	function deleteTodo(id){
+	const taskElement = document.querySelector(`li[id='${id}']`);
+	taskElement.remove();
+	}
+	
+	
+	function modifyTodo(event){
+	const button = event.target;
+	const taskid = button.getAttribute('taskid');
+	
+	const todoItem = button.parentNode.parentNode;
+	const todoTitle = todoItem.firstChild.textContent.trim();
+
+  const inputField = document.createElement('input');
+  inputField.type = 'text';
+  inputField.value = todoTitle;
+  const saveButton = document.createElement('button');
+  saveButton.textContent = 'Save';
+  saveButton.addEventListener("click", function (event) {
+    const newTitle = inputField.value.trim();
+    if (newTitle !== '') {
+      todoItem.firstChild.textContent = newTitle;
+      todoItem.removeChild(inputField);
+      todoItem.removeChild(saveButton);
+    } else {
+      alert('Please enter a valid title!');
+    }
+    fetch(`http://localhost:3000/todos/:${todoItem.id}`, {
+      method: "PATCH", 
+      headers: {
+        "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+        id: +todoItem.id,
+        title: newTitle})
+      }).then(res => {
+          console.log(res.status);
+        })
+    });
+  
+    todoItem.appendChild(inputField);
+    todoItem.appendChild(saveButton);
+	}
+	
+	function removeCompletedTodos() {
+  const completedTodos = document.querySelectorAll('.completed');
+  completedTodos.forEach(todo => todo.remove());
+}
+
+
+document.querySelector('#newTodoContainer button:nth-of-type(1)').addEventListener('click', function () {
+  fetch("http://localhost:3000/todos", {
+      method: "POST", 
+      headers: {
+          "Content-Type": "application/json",
+      },
+      body: JSON.stringify({
+          title: document.getElementById('newTodoInput').value,
+          done: false})
+  }).then(res => {
+    const todoItem = document.getElementById('newTodoInput');
+    const todosContainer = document.querySelector("ul");
+    const liCount = todosContainer.querySelectorAll('li').length;
+    if (todoItem.value.trim() !== '') {
+      let item = {
+        title: todoItem.value.trim(),
+        done: false,
+        id: liCount + 1
+      };
+      addTodo(item);
+      document.getElementById('newTodoInput').value = " ";
+      console.log(res);
+    } else {
+      alert('Please enter a todo title!');
+    }
+  })
+
+});
+
+document.querySelector('#newTodoContainer button:nth-of-type(2)').addEventListener('click', function () {
+fetch("http://localhost:3000/todos", {
+      method: "PUT", 

```

Comment:
because you are binding "deleting" functionality to the button here, you should use REST method "DELETE"
that request to the server can be sent to either this path `DELETE /todos/:id` - for deleting only task with the specific `id`, of you can could do it more complex and also send data in the request body with a list if IDs to delete. But for now just choose the `DELETE /todos/:id` one,

Reply:
oh, and obviously, `DELETE /todos/:id` route should be implemented at the server side.